python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -261,15 +261,21 @@ class NearestNeighbors(Base):
             Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
             ndarray, cuda array interface compliant array like CuPy
 
+        convert_dtype : bool, optional (default = True)
+            When set to True, the fit method will automatically
+            convert the inputs to np.float32.
 
         """
 
         if self._should_downcast:
             warnings.warn("Parameter should_downcast is deprecated, use "
-                          "convert_dtype in fit, fit_transform and transform "
+                          "convert_dtype in fit and kneighbors "
                           " methods instead. ")
             convert_dtype = True
 
+        if self._should_downcast is None:
+            self._should_downcast = True
```

Comment:
As below, it's better to handle this self._should_downcast param in the initializer not in the fit or predict functions. Then it's only set in one place and treated as a real parameter.